pkg/securitycontext/provider.go

```diff
@@ -65,7 +65,9 @@ func (p SimpleSecurityContextProvider) ModifyHostConfig(pod *v1.Pod, container *
 			for _, group := range pod.Spec.SecurityContext.SupplementalGroups {
 				hostConfig.GroupAdd = append(hostConfig.GroupAdd, strconv.Itoa(int(group)))
 			}
-			if pod.Spec.SecurityContext.FSGroup != nil {
+			// only add the FSGroup to the config if we are not in a remap environment.  For remap environments
+			// is is assumed that the root user is remapped to the FSGroup on the host.
+			if pod.Spec.SecurityContext.FSGroup != nil && !isRemap {
```

Comment:
ie, not remapping groups at all and only users?  Currently, you have to have a matching remap gid in /etc/subgid for your --userns-remap argument so I don't think that is the case.  If you do not have the group in /etc/subgid you'll currently get a daemon error and it will fail to start

```
level=fatal msg="Error starting daemon: Can't create ID mappings: No subgid ranges found for group \"dockremap\""
```